gui.py

```diff
@@ -12,99 +14,169 @@ def __init__(self, parent=None):
+        QtCore.QObject.connect(self.manager.emitter, QtCore.SIGNAL('building(QString, int, int)'), self.update_upload_btn)
+
+        # If bad path at launch, pop up the filedialog to locate it
+        if not os.path.exists(self.game_dir):
+            QtGui.QMessageBox.warning(self, "Game directory not found", "Please set your game directory")
+            self.search_game_directory()
+
+    def get_user_data(self):
+        if os.path.exists('userdata.json'):
+            need_clear = False
+            with open('userdata.json', 'r') as f:
+                try:
+                    self.user_data = json.load(f)
+                    print(self.user_data)
+                except:
+                    need_clear = True
+                    print(os.getcwd())
+            if need_clear:
+                os.remove('userdata.json')
+
+    def dump_user_data(self):
+        with open('userdata.json', 'w') as f:
+            json.dump(self.user_data, f)
+
+    def update_upload_btn(self, text, val, total):
+        if total > 0:
+            percent = int(float(val)/total*100)
+            self.upload_btn.setText("{} {}%".format(text, percent))
+        else:
+            self.upload_btn.setText("{}".format(text))
 
+        self.repaint()
+
+    def set_upload_btn_state(self, state=None):
+        ''' Update the upload btn status '''
+        if state == 'upload':
+            self.upload_btn.setText('Uploading...')
+            self.upload_btn.setStyleSheet("background-color: rgb(28, 170, 217); color: white;")
+        elif state == 'building':
+            self.upload_btn.setEnabled(True)
+            self.upload_btn.setText('Building zip archive...')
+            self.upload_btn.setStyleSheet("background-color: rgb(170, 170, 170); color: white;")
+        elif state =='publish':
+            self.upload_btn.setEnabled(True)
+            self.upload_btn.setText('Publish to Sketchfab')
+            self.upload_btn.setStyleSheet("background-color: rgb(28, 170, 217); color: white;")
+
+        self.repaint()
+
+    def description_input_limit(self):
+        if len(self.description_tb.toPlainText()) > 1024:
+            self.description_tb.setText(self.description_tb.toPlainText()[:1024])
+        self.description_label.setText('Description ({}/1024)'.format(len(self.description_tb.toPlainText())))
 
     def updateList(self, index):
+        ''' Update Gui when list element is changed'''
         self.upload_btn.setEnabled(True)
-        self.upload_btn.setText('Upload {}'.format(self.craft_list[index].split('.')[1].strip()))
+        self.upload_btn.setText('Publish to Sketchfab')
         self.status_info_label.setText('')
-
-    def update_window_data(self):
-        if os.path.exists(self.game_dir):
-            self.manager = KSP2Skfb(self.game_dir)
-            self.craft_list = self.manager.get_craft_list()
+        self.enable_skfb_ui(True)
+        self.name_tb.setText('{}'.format(self.craft_list[index].split('.')[1].strip()))
```

Comment:
it's difficult to understand what `self.craft_list[index].split('.')[1].strip())` does here; would probably be better in a rightly named function